### ISSUE TITLE
[R-package] Getting rid of default metric without custom loss function

### ISSUE THREAD

**Issue by Laurae2** [Microsoft/LightGBM#1821]
Currently, LightGBM forces to use the default objective loss in addition to the custom loss function in R. This can be seen using the following (taken and adapted from https://github.com/Microsoft/LightGBM/blob/master/R-package/demo/early_stopping.R):

```r
require(lightgbm)
require(methods)

# Load in the agaricus dataset
data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")

dtrain <- lgb.Dataset(agaricus.train$data, label = agaricus.train$label)
dtest <- lgb.Dataset(agaricus.test$data, label = agaricus.test$label)

param <- list(num_leaves = 4,
              learning_rate = 1)
valids <- list(eval = dtest)
num_round <- 20

evalerror <- function(preds, dtrain) {
  labels <- getinfo(dtrain, "label")
  err <- as.numeric(sum(labels != (preds > 0.5))) / length(labels)
  return(list(name = "error", value = err, higher_better = FALSE))
}

bst <- lgb.train(param,
                 dtrain,
                 num_round,
                 valids,
                 objective = "binary",
                 eval = evalerror,
                 early_stopping_round = 3)
```

Leads to the following (unwanted binary_logloss):

```
[LightGBM] [Warning] Starting from the 2.1.2 version, default value for the "boost_from_average" parameter in "binary" objective is true.
This may cause significantly different results comparing to the previous versions of LightGBM.
Try to set boost_from_average=false, if your old models produce bad results
[LightGBM] [Info] Number of positive: 3140, number of negative: 3373
[LightGBM] [Info] Total Bins 128
[LightGBM] [Info] Number of data: 6513, number of used features: 107
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.482113 -> initscore=-0.071580
[LightGBM] [Info] Start training from score -0.071580
[1]:	eval's binary_logloss:0.204754	eval's error:0.0335196 
[2]:	eval's binary_logloss:0.113096	eval's error:0.0217256 
[3]:	eval's binary_logloss:0.0476511	eval's error:0.00558659 
[4]:	eval's binary_logloss:0.0234622	eval's error:0 
[5]:	eval's binary_logloss:0.11908	eval's error:0.00372439 
[6]:	eval's binary_logloss:0.0154869	eval's error:0 
[7]:	eval's binary_logloss:0.00882988	eval's error:0 
```

A workaround is the following, but this is not obvious and is probably an unexpected case:

```r
bst <- lgb.train(param,
                 dtrain,
                 num_round,
                 valids,
                 objective = "binary",
                 eval = evalerror,
                 metric = 0,
                 early_stopping_round = 3)
```

```
[LightGBM] [Warning] Starting from the 2.1.2 version, default value for the "boost_from_average" parameter in "binary" objective is true.
This may cause significantly different results comparing to the previous versions of LightGBM.
Try to set boost_from_average=false, if your old models produce bad results
[LightGBM] [Info] Number of positive: 3140, number of negative: 3373
[LightGBM] [Info] Total Bins 128
[LightGBM] [Info] Number of data: 6513, number of used features: 107
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.482113 -> initscore=-0.071580
[LightGBM] [Info] Start training from score -0.071580
[1]:	eval's error:0.0335196 
[2]:	eval's error:0.0217256 
[3]:	eval's error:0.00558659 
[4]:	eval's error:0 
[5]:	eval's error:0.00372439 
[6]:	eval's error:0 
[7]:	eval's error:0 
```

**Comment by guolinke** [comment]
refe to https://lightgbm.readthedocs.io/en/latest/Python-API.html#lightgbm.train

> ... To ignore the default metric corresponding to the used objective, set the metric parameter to the string "None" in params.

**Comment by Laurae2** [comment]
@guolinke Seems to work correctly with "None". "0" also seems to work but is probably an unexpected case.
